HousingManagementSystemApi.Tests/UseCasesTests/VerifyPropertyEligibilityUseCaseTests.cs

```diff
@@ -0,0 +1,196 @@
+                .ReturnsAsync((TenureInformation)null);
+
+
+            // Act
+            var result = await this.sut.Execute(HouseThatExists);
+
+            // Assert
+            Assert.False(result.PropertyEligible);
+            Assert.Contains("Tenure type code was null for asset", result.Reason);
+        }
+
+        [Fact]
+        public async Task GivenAssetWithAnInvalidTenureType_WhenUseCaseIsExecuted_ThenThereWillBeAFailureResult()
+        {
+            // Arrange
+            const string HouseThatExists = "01234567";
+
+            retrieveAssetGateway.Setup(x => x.RetrieveAsset(HouseThatExists))
+                .ReturnsAsync(new AssetResponseObject
+                {
+                    AssetType = AssetType.Dwelling,
+                    Tenure = new AssetTenureResponseObject
+                    {
+                        Id = "TEN001"
+                    }
+                });
+
+            tenureGateway.Setup(x => x.RetrieveTenureType("TEN001"))
+                .ReturnsAsync(new TenureInformation { TenureType = TenureTypes.AsylumSeeker });
+
+            // Act
+            var result = await this.sut.Execute(HouseThatExists);
+
+            // Assert
+            Assert.False(result.PropertyEligible);
+            Assert.Contains("is not suitable for Online Repairs", result.Reason);
+        }
+
+        [Fact]
+        public async Task GivenAllValidData_WhenUseCaseIsExecuted_ThenShouldBeASuccessfulValidation()
+        {
+            // Arrange
+            const string HouseThatExists = "01234567";
+
+            retrieveAssetGateway.Setup(x => x.RetrieveAsset(HouseThatExists))
+                .ReturnsAsync(new AssetResponseObject
+                {
+                    AssetType = AssetType.Dwelling,
+                    Tenure = new AssetTenureResponseObject
+                    {
+                        Id = "TEN001"
+                    }
+                });
+
+            tenureGateway.Setup(x => x.RetrieveTenureType("TEN001"))
+                .ReturnsAsync(new TenureInformation { TenureType = TenureTypes.Secure });
+
+            // Act
+            var result = await this.sut.Execute(HouseThatExists);
+
+            // Assert
+            Assert.True(result.PropertyEligible);
+        }
+
+        [Fact]
+        public async Task GivenAPostcode_WhenAnAddressExistsWithANullTenure_ThenAPropertyIsNotReturned()
+        {
+            const string TestPostcode = "postcode";
+            var eligibleTenureType = TenureTypes.Secure;
+
+            retrieveAssetGateway.Setup(x => x.RetrieveAsset("assetId"))
+                .ReturnsAsync(new AssetResponseObject { AssetType = AssetType.Dwelling });
+
+
+            tenureGateway.Setup(x => x.RetrieveTenureType("Id"))
+                .ReturnsAsync(new TenureInformation { TenureType = TenureTypes.Secure });
+
+            // Act
+            var result = await this.sut.Execute(TestPostcode);
+        }

```

Comment:
No assertion? Also why is a postcode being passed into `VerifyPropertyEligibilityUseCase.Execute(postcode)` ?

Reply:
This was a test left over from another file